botocore/credentials.py

```diff
@@ -1253,8 +1288,35 @@ def _load_creds_via_assume_role(self, profile_name):
             refresh_using=refresher,
             time_fetcher=_local_now
         )
+    
+    def _get_role_config_from_role_arn(self, role_arn, profile_name=None):
+        """Retrieves and validates the role configuration from a specified role_arn."""
+        if profile_name is not None:
+            source_profile = profile_name
+            credential_source = None
+        else:
+            source_profile = None
+            credential_source = 'auto' # TODO
```

Comment:
HACK:
currently AssumeRoleProvider is written with the assumption that the credential_source is always set, and that it should fail if it is incorrect. This PR requires that we do some guesswork, though. By doing this, I can use the same _validate(role_config) path for both the _from_role_arn and _from_profile() paths, which is really nice. Can you think of a way to modify AssumeRoleProvider in a nice way such that we don't go duplicating all the validation logic? Thoughts?